xls/dslx/stdlib/std.x

```diff
@@ -396,6 +396,41 @@ fn round_up_to_nearest_test() {
     assert_eq(u32:4, round_up_to_nearest(u32:4, u32:2));
 }
 
+// Returns `x` rounded up to `y`, where `y` is a positive power of 2
+fn round_up_to_pow2_unsigned<N: u32>(x: uN[N], y: uN[N]) -> uN[N] { (x + y - uN[N]:1) & -y }
+
+#[test]
+fn test_round_up_to_pow2_unsigned() {
+    assert_eq(round_up_to_pow2_unsigned(u16:0, u16:8), u16:0);
+    assert_eq(round_up_to_pow2_unsigned(u16:1, u16:8), u16:8);
+    assert_eq(round_up_to_pow2_unsigned(u16:7, u16:8), u16:8);
+    assert_eq(round_up_to_pow2_unsigned(u16:8, u16:8), u16:8);
+    assert_eq(round_up_to_pow2_unsigned(u16:9, u16:8), u16:16);
+
+    assert_eq(round_up_to_pow2_unsigned(u16:9, u16:16), u16:16);
+}
+
+// Returns `x` rounded up to `y`, where `y` is a positive power of 2
+fn round_up_to_pow2_signed<N: u32>(x: sN[N], y: uN[N]) -> sN[N] {
+    (x + y as sN[N] - sN[N]:1) & -(y as sN[N])
+}
+
+#[test]
+fn test_round_up_to_pow2_signed() {
+    assert_eq(round_up_to_pow2_signed(s16:9, u16:8), s16:16);
```

Comment:
should this be rounded to `8`?

Reply:
Sorry for the confusion. The function should do the same as the existing `std::round_up_to_nearest()`, but generate much simpler logic.